### DISCUSSION TITLE
Support for Google Optimize Anti-Flicker Snippet in gatsby-plugin-google-gtag

### DISCUSSION THREAD

**Discussion by joelwilsonmt** [gatsbyjs/gatsby#24121] [Ideas / Feature Requests]
## Summary

When using gatsby-plugin-google-gtag it would be nice to have the google optimize anti-flicker snippet added to the top of the document as per [Google's installation guidelines](https://support.google.com/optimize/answer/7100284?hl=en)

### Basic example

Change the [gatsby-ssr.js file](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-google-gtag/src/gatsby-ssr.js) to include the above snippet (CSS and JS), which can then be passed to `setComponents()` on line 79. I don't know if this will place the snippet in the right place - for example if using React Helmet, will it place it top-most in the `<head>` of the document?

### Motivation

To be able to have out-of-the-box support for the Google Optimize Anti-flicker snippet. At the moment Optimize Experiences are delivered with a screen flicker as elements are replaced on the page as the page loads.

**Comment by Odelya** [comment]
For best performance, the snippet that loads the Optimize plugin (whether it's Analytics, Tag Manager, or gtag.js) should be installed in the <HEAD> element of your web page, immediately after the anti-flicker snippet . The Deploy Optimize help center article and tagging best practices section contain more information on the best way to install Optimize.
